client/src/plugins/zeebe-plugin/deployment-plugin/DeploymentPlugin.js

```diff
@@ -145,11 +147,18 @@ export default class DeploymentPlugin extends PureComponent {
     // (4) deploy
     const deploymentResult = await this.deployWithConfig(savedTab, config);
 
-    // (5) notify interested parties
+    // (5) fetch version deployed as contextual information
+    const getTopologyResult = await this.getTopology(config);
+
+    const { gatewayVersion } = getTopologyResult.response;
+    options.gatewayVersion = gatewayVersion;
```

Comment:
I had another look at this today and I think that the deployment plugin does not need to know anything about topology. I'd rather expect that we call the backend to `getEngineVersion` and expect to get the engine version as a result. So no topology, no gateway version as these are the implementation details of Zeebe API.

Reply:
So if the names change in the future, we touch only `zeebe-api` on the backend and not both the backend and the client.

Reply:
```suggestion
    // (5) fetch version deployed as contextual information
    options.engineVersion = await this.getEngineVersion(config);
```

Reply:
I see the point.

I adjusted it in https://github.com/camunda/camunda-modeler/pull/2237/commits/b1260214ba1710a2bb2bac816eb0797169e13fb1.

I am not 100% sure here (but it is also not critical):

* Now `zeebe-api.js` is not only a rather stupid wrapper around the `zeebe-node-api`, but it actually extracts information from it. This is what I don't like
* As an alternative 1 we should keep zeebe-api#getTopolgy and wrap around that again. However, that increases the codeBase, while not adding so much value
* (Or we go back to the initial solution without https://github.com/camunda/camunda-modeler/pull/2237/commits/b1260214ba1710a2bb2bac816eb0797169e13fb1)

At the end of the day: with and without https://github.com/camunda/camunda-modeler/pull/2237/commits/b1260214ba1710a2bb2bac816eb0797169e13fb1 the solution is good from my perspective. Happy to discuss if you want.

Reply:
I like the new solution more. I still don't know what "gateway" is, but I think it's overall much better than having just a stupid wrapper. Note that we already interpret the errors in `zeebe-api.js`.